index.html

```diff
@@ -101,212 +220,1263 @@ <h3>
               The port that the socket should be bound to. When omitted, or 0,
               any port can be used.
             </dd>
+            <div class="note">
+              The only current implementation (Chromium) ignores
+              {{TCPSocketOptions/localAddress}} and {{TCPSocketOptions/localPort}}.
+            </div>
+            -->
+            <dt>
+              <dfn>noDelay</dfn> member
+            </dt>
+            <dd>
+              Enables the `TCP_NODELAY` option,
+              disabling <a href=
+              "https://en.wikipedia.org/wiki/Nagle%27s_algorithm">Nagle's
+              algorithm</a>.
+            </dd>
+            <dt>
+              <dfn>keepAliveDelay</dfn> member
+            </dt>
+            <dd>
+              Enables TCP Keep-Alive by setting `SO_KEEPALIVE` option on the socket equal
+              to the specified Keep-Alive ping delay in milliseconds.
+              <div class="note">
+                Keep-Alive is disabled by default.
+              </div>
+            </dd>
+          </dl>
+
+        </section>
+
+      </section>
+
+      <section>
+        <h3>{{TCPSocket/[[readable]]}} attribute (internal)</h3>
+
+        <aside class="example">
+          An application receiving data from a TCP socket will typically use the {{ReadableStream}} like this:
+
+          <pre class="js">
+            const { readable } = await socket.opened;
+
+            const reader = readable.getReader();
+
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) {
+                // |reader| has been canceled.
+                break;
+              }
+              // Do something with |value|...
+            }
+
+            reader.releaseLock();
+          </pre>
+        </aside>
+      
+        The steps to <dfn data-lt="initialize tcp readable stream">initialize the TCP readable stream</dfn> are:
+
+        <ol>
+          <li>
+            Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Run the following steps in parallel:
+              <ol>
+                <li>
+                  Invoke the operating system to read a tcp packet from the socket, placing the result in the [=byte sequence=] |bytes|.
+                <li>
+                  [=Queue a global task=] on the [=relevant global object=] of
+                  [=this=] using the [=TCP socket task source=] to run the
+                  following steps:
+                  <ol>
+                    <li>
+                      If FIN packet was received, run the following steps:
```

Comment:
Similarly this can simply say "the connection was closed gracefully". Maybe we could add a note like "this is triggered by the peer sending a packet with the FIN flag set and is typically indicated by the operating system returning 0 bytes when asked for more data from the socket."

Reply:
Fixed!